src/storage/gpu_swap_prefetch.cc

```diff
@@ -129,21 +124,21 @@ void Prefetch::PrefetchWhileComputing(int device) {
 
 }
 
-
 // TODO(sotskin): karll: Add algorithm discription
 void Prefetch::HistoryBasedPrefetch(int device) {
   //pthread_rwlock_rdlock(&swap_lock_);
   //bool has_begun = false;
-  if(lookahead_pos_[device] < (int)history_->record_idx[device]) {
-    lookahead_pos_[device] = history_->record_idx[device];
+  auto& history = history_->DevHistory(device);
+  if (lookahead_pos_[device] < history.curr_idx) {
+    lookahead_pos_[device] = history.curr_idx;
   }
-  while(lookahead_pos_[device]+1 < history_->ordered_history[device].size() &&
-      lookahead_pos_[device] - (int)history_->record_idx[device] < steps_ahead_ &&
-      lookahead_pos_[device] - (int)history_->record_idx[device] >= 0 ) {
-    MemHistory::MemRecord r =
-        history_->ordered_history[device][++lookahead_pos_[device]];
-    if(r.operation_id == MemHistory::GET_ADDR) {
-      ++(history_->prefetch_count);
+  while (lookahead_pos_[device] + 1 < history.ordered_history.size() &&
+      lookahead_pos_[device] - history.curr_idx < steps_ahead_ &&
+      lookahead_pos_[device] - history.curr_idx >= 0 ) {
```

Comment:
This condition may cause overflow because lookahead_pos_ is a vector of int while curr_idx is size_t, unsigned int. I suggest we change lookahead_pos_ to a vector of size_t and check the condition (lookahead_pos_[device] >= history.curr_idx) instead.

Reply:
Good point. Will change it.